### PR TITLE
Add skybox .h and cleanup, `SkyboxDraw_` -> `Skybox_`

### PR DESCRIPTION
**Pull request by Dragorn421**

### CODE REVIEW

File: include/z64skybox.h [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+typedef struct {
+    /* 0x00 */ RomFile file;
+    /* 0x08 */ RomFile palette;
+} SkyboxFile; // size = 0x10
+
+extern SkyboxFile gNormalSkyFiles[];
+
+struct GameState;
+struct GraphicsContext;
```
Comment (AngheloAlf) [inline include/z64skybox.h:61]: Personally I like putting this kind of forward declarations at the top of the file, but I think it is fine either way

**Review by AngheloAlf — APPROVED** [review]

File: include/z64skybox.h (hunk 1 above)
Comment (Dragorn421) [reply include/z64skybox.h:61]: Sure, I don't mind

**Review by engineer124 — APPROVED** [review]

File: include/z64skybox.h [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    /* 0x08 */ RomFile palette;
+} SkyboxFile; // size = 0x10
+
+extern SkyboxFile gNormalSkyFiles[];
+
+void Skybox_Init(struct GameState* state, SkyboxContext* skyboxCtx, s16 skyboxId);
+Mtx* SkyboxDraw_UpdateMatrix(SkyboxContext* skyboxCtx, f32 x, f32 y, f32 z);
+void SkyboxDraw_Draw(SkyboxContext* skyboxCtx, struct GraphicsContext* gfxCtx, s16 skyboxId, s16 blend, f32 x, f32 y,
+                     f32 z);
+void SkyboxDraw_Update(SkyboxContext* skyboxCtx);
```
Comment (fig02) [inline include/z64skybox.h:67]: doesnt have to happen in this pr, but we can consider making these just Skybox_
they are in a different file, but it doesnt really make it a different system imo. and skybox draw draw sounds dumb lol

**Review by fig02 — APPROVED** [review]

File: include/z64skybox.h (hunk 2 above)
Comment (fig02) [reply include/z64skybox.h:67]: to elaborate on my thoughts, everything in vr_box.c runs once on init, so the draw and (empty) update stuff being in another file is purely an organizational thing. its not a separate system with its own update/draw

the update is only empty cause environment already handles everything lol `Environment_UpdateSkybox`, which I think can still coexist with `Skybox_Update`

Comment (fig02) [reply include/z64skybox.h:67]: actually im very on board with this and think we should make the change now lol

Comment (engineer124) [reply include/z64skybox.h:67]: I've recently been working on skybox through MM, and I'm also fine with this change

**Review by fig02 — APPROVED** [review]

**Review by engineer124 — APPROVED** [review]

**Pull request merged**